### PR TITLE
sui-node: introduce admin interface

### PR DESCRIPTION
**Pull request by bmwill**
Introduce an admin interface, bound to localhost, that enables dynamically reconfiguring the logging level at runtime.

```
# Query current logging filter
curl http://localhost:<port>/logging

# Set a new logging filter
curl http://localhost:<port>/logging -d "debug,hyper=off"
```

Closes: #3178

### CODE REVIEW

File: crates/sui-config/src/node.rs [hunk 1]
```diff
@@ -59,6 +62,10 @@ fn default_metrics_address() -> SocketAddr {
     SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), 9184)
 }
 
+pub fn default_admin_interface_port() -> u16 {
+    1337
```
Comment (allan-bailey) [inline crates/sui-config/src/node.rs:66]: why is this a function call instead of a config value?

**Review by allan-bailey — APPROVED** [review]
LGTM. just the 1 question.

File: crates/sui-node/src/admin.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
```
Comment (huitseeker) [inline crates/sui-node/src/admin.rs:3]: This is kind of ad-hoc, so I'd add a line or two for what we want this to accomplish.

File: crates/sui-config/src/node.rs (hunk 1 above)
Comment (bmwill) [reply crates/sui-config/src/node.rs:66]: It is a config value, this is just the default in the event it wasn't specified in the config

File: crates/sui-node/src/admin.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        Err(err) => (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()),
+    }
+}
+
+async fn set_filter(
+    Extension(filter_handle): Extension<FilterHandle>,
+    new_filter: String,
+) -> (StatusCode, String) {
+    match filter_handle.update(new_filter) {
+        Ok(()) => (StatusCode::OK, "".into()),
```
Comment (huitseeker) [inline crates/sui-node/src/admin.rs:45]: Perversely enough, would it be possible to add something like `info!("Log EnvFilter string changed to {new_filter}`, I have a feeling that future me, equipped with just a console and no fancy tooling, will appreciate having this anchor in logs as a starting point.

Plus it serves as a trace of what happened in the admin interface: I can envision someone exposing this to the exterior, and a kid passing by discovering they can kill the node my hitting that endpoint with the word "trace".

Comment (bmwill) [reply crates/sui-node/src/admin.rs:45]: > I can envision someone exposing this to the exterior, and a kid passing by discovering they can kill the node my hitting that endpoint with the word "trace".

Thankfully the code prohibits this as you need to be on localhost to hit it

**Pull request merged**